AutomatedTesting/Gem/PythonTests/Prefab/tests/duplicate_prefab/DuplicatePrefab_ContainingNestedEntitiesAndNestedPrefabs.py

```diff
@@ -0,0 +1,110 @@
+    from editor_python_test_tools.prefab_utils import Prefab, get_all_entity_ids, wait_for_propagation
+    import Prefab.tests.PrefabTestUtils as prefab_test_utils
+
+    NESTED_ENTITIES_PREFAB_FILE_NAME = Path(__file__).stem + '_' + 'nested_entities_prefab'
+    NESTED_ENTITIES_NAME_PREFIX = 'Entity_'
+    NESTED_PREFABS_FILE_NAME_PREFIX = Path(__file__).stem + '_' + 'nested_prefabs_'
+    NESTED_PREFABS_NAME_PREFIX = 'NestedPrefabs_Prefab_'
+    FILE_NAME_OF_PREFAB_WITH_NESTED_ENTITIES_AND_NESTED_PREFABS = Path(__file__).stem + '_' + 'new_prefab'
+    NESTED_PREFABS_TEST_ENTITY_NAME = 'TestEntity'
+    PHYSX_COLLIDER_NAME = 'PhysX Collider'
+    CREATION_POSITION = azlmbr.math.Vector3(100.0, 100.0, 100.0)
+    NUM_NESTED_ENTITIES_LEVELS = 3
+    NUM_NESTED_PREFABS_LEVELS = 3
+
+    prefab_test_utils.open_base_tests_level()
+
+    # Creates new nested entities at the root level
+    # Asserts if creation didn't succeed
+    nested_entities_root = prefab_test_utils.create_linear_nested_entities(
+        NESTED_ENTITIES_NAME_PREFIX, NUM_NESTED_ENTITIES_LEVELS, CREATION_POSITION)
+    prefab_test_utils.validate_linear_nested_entities(nested_entities_root, NUM_NESTED_ENTITIES_LEVELS,
+                                                      CREATION_POSITION)
+    nested_entities_root_name = nested_entities_root.get_name()
+
+    # Creates new nested prefabs at the root level
+    # Asserts if creation didn't succeed
+    entity_to_nest = EditorEntity.create_editor_entity_at(CREATION_POSITION, name=NESTED_PREFABS_TEST_ENTITY_NAME)
+    assert entity_to_nest.id.IsValid(), "Couldn't create TestEntity"
+    entity_to_nest.add_component(PHYSX_COLLIDER_NAME)
+    assert entity_to_nest.has_component(PHYSX_COLLIDER_NAME), f"Failed to add a {PHYSX_COLLIDER_NAME}"
+    _, nested_prefab_instances = prefab_test_utils.create_linear_nested_prefabs(
+        [entity_to_nest], NESTED_PREFABS_FILE_NAME_PREFIX, NESTED_PREFABS_NAME_PREFIX, NUM_NESTED_PREFABS_LEVELS)
+    prefab_test_utils.validate_linear_nested_prefab_instances_hierarchy(nested_prefab_instances)
+
+    # Creates a new prefab containing the nested entities and the nested prefab instances
+    # Asserts if prefab creation doesn't succeed
+    _, new_prefab = Prefab.create_prefab(
+        [nested_entities_root, nested_prefab_instances[0].container_entity],
+        FILE_NAME_OF_PREFAB_WITH_NESTED_ENTITIES_AND_NESTED_PREFABS)
+    new_prefab_container_entity = new_prefab.container_entity
+
+    nested_entities_root_on_instance = new_prefab.get_direct_child_entities()[0]
+    nested_prefabs_root_container_entity_on_instance = new_prefab.get_direct_child_entities()[1]
+    if new_prefab.get_direct_child_entities()[0].get_name() != nested_entities_root_name:
+        nested_entities_root_on_instance, nested_prefabs_root_container_entity_on_instance = \
+            nested_prefabs_root_container_entity_on_instance, nested_entities_root_on_instance
```

Comment:
This answers my previous question I guess. Rather than swap variables like this, writing a small helper function to fetch index of entity based on name would be a cleaner way to get entities when order is unknown.

Reply:
Done.